Use cisagov/session-manager-tf-module to configure SSM Session Manager

jsf9k opened the pull request:
## 🗣 Description ##

This pull request makes use of [cisagov/session-manager-tf-module](https://github.com/cisagov/session-manager-tf-module) to set up access for and configure SSM Session Manager.

## 💭 Motivation and context ##

We want to:
- Stop using ssh tunneling over SSM
- Start logging SSM sessions
- DRY out some repeated code

## 🧪 Testing ##

These changes have been successfully applied to the staging COOL environment.

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All relevant repo and/or project documentation has been updated to reflect the changes in this PR.
- [x] All new and existing tests pass.

## ✅ Pre-merge checklist ##

- [x] Revert dependencies to default branches.

jsf9k committed:
Use cisagov/session-manager-tf-module to configure SSM Session Manager

jsf9k committed:
Update README via terraform-docs

dav3r commented on shared-services/outputs.tf:
```diff
@@ -9,6 +9,6 @@ output "provisionaccount_role" {
 output "ssmsession_role" {
-  value       = aws_iam_role.ssmsession_role
+  value       = module.session_manager.ssm_session_role
   description = "The IAM role that allows creation of SSM SessionManager sessions to any EC2 instance in this account."
```
So this is where it all started.....
```suggestion
  description = "The IAM role that allows creation of SSM Session Manager sessions to any EC2 instance in this account."
```

jsf9k replied:
Please see commit 9264e5f.

dav3r reviewed (commented):
This looks solid to me. 👍 👍  
Go forth and conquer the rest of the accounts in this repo. 🚀

jsf9k committed:
Treat Session Manager as two words, like AWS does

Co-authored-by: dav3r <david.redmin@trio.dhs.gov>

jsf9k commented:
> Go forth and conquer the rest of the accounts in this repo.

Given that we decided to only add this jazz to accounts that actually have EC2 instances running in them, no other accounts require these changes.  One could argue for adding them to the dynamic accounts, but given the aforementioned decision it makes more sense to add them _if_ the dynamic account will actually contain EC2 instances; for example, I will add these changes to the [cisagov/cool-assessment-terraform](https://github.com/cisagov/cool-assessment-terraform) repository.

dav3r reviewed (approved):
Fantastic! 💎

jsf9k committed:
Add more permissions to policy for provisioning the SSM Session Manager

These permissions were found to be missing when working on cisagov/cool-assessment-terraform#186.

jsf9k committed:
Revert to using the default branch of cisagov/session-manager-tf-module

The branch that we were using was merged into the default branch via
cisagov/session-manager-tf-module#4.

mcdonnnj commented on shared-services/provisionssmsessionmanager_policy_attachment.tf:
```diff
@@ -1,9 +1,10 @@
 # ------------------------------------------------------------------------------
 # Attach to the ProvisionAccount role the IAM policy that allows
-# provisioning of the SSM Document resource required in this account.
+# provisioning of the SSM Document resource and set up SSM session
+# logging in this account.
```
Just to align with the comment in `shared-services/provisionssmsessionmanager_policy.tf`. What drew my attention here was

```
# provisioning of the SSM Document resource and set up SSM session
```

so if you don't like this wholesale replacement I would at least like the `set` above changed.

```suggestion
# all of the permissions necessary to provision the SSM Document
# resource and set up SSM session logging in this account.
```

jsf9k replied:
Please see commit 1bd322c.

mcdonnnj reviewed (approved):
Nice improvement. I had one suggestion related to a comment.

jsf9k committed:
Update comment to parallel similar comment in shared-services/provisionssmsessionmanager_policy.tf

Co-authored-by: Nick <50747025+mcdonnnj@users.noreply.github.com>
